components/profile/layout/MissingNameModal/MissingNameModal.tsx

```diff
@@ -0,0 +1,68 @@
+import { useContext, useEffect, useState } from "react";
+import { DonorContext } from "../donorProvider";
+import { Lightbox } from "../../../shared/components/Lightbox/Lightbox";
+import { EffektTextInput } from "../../../shared/components/EffektTextInput/EffektTextInput";
+import { saveDonor } from "../../_queries";
+import { useAuth0 } from "@auth0/auth0-react";
+import { AlertCircle, Check } from "react-feather";
+import { toast } from "react-toastify";
+
+import styles from "./MissingNameModal.module.scss";
+
+export const MissingNameModal: React.FC = () => {
+  const { donor, setDonor } = useContext(DonorContext);
+  const { user, getAccessTokenSilently } = useAuth0();
+  const [name, setName] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [open, setOpen] = useState(false);
+
+  useEffect(() => {
+    if (donor && !donor.name) {
+      setOpen(true);
+    }
+  }, [donor]);
```

Comment:
Not sure if it's a big deal, but I slightly suspect the modal might reopen each time this rerenders if the user for some reason don't want to give their name by cancelling the modal. Perhaps something to just check manually, if it's any concern. Otherwise, looks great! :+1:

Reply:
That was the idea, let's bug the user to give us their name whenever they log in :)

Reply:
I did consider just not allowing them to close the modal at all, but thought it was a little bit restrictive